service/controller/v14/adapter/adapter_key.go

```diff
@@ -0,0 +1,31 @@
+package adapter
+
+import "github.com/giantswarm/aws-operator/service/controller/v14/key"
+
+func asgType(config Config) string {
+	return prefixWorker
+}
+
+func baseDomain(config Config) string {
+	return key.BaseDomain(config.CustomObject)
+}
+
+func clusterID(config Config) string {
+	return key.ClusterID(config.CustomObject)
+}
+
+func hostedZoneNameServers(config Config) string {
+	return config.StackState.HostedZoneNameServers
+}
+
+func masterInstanceResourceName(config Config) string {
+	return config.StackState.MasterInstanceResourceName
+}
+
+func route53Enabled(config Config) bool {
+	return config.Route53Enabled
+}
+
+func workerImageID(config Config) string {
+	return config.StackState.WorkerImageID
+}
```

Comment:
I extracted `adapter_key.go`, because there are repeated extractions. Ultimate goal would be to have all extractions in here like in `key/key.go` file.